.pre-commit-config.yaml

```diff
@@ -0,0 +1,48 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: check-added-large-files
+      - id: check-case-conflict
+      - id: check-merge-conflict
+      - id: check-symlinks
+      - id: check-toml
+      - id: debug-statements
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+  - repo: https://github.com/pre-commit/pygrep-hooks
+    rev: v1.9.0
+    hooks:
+      - id: python-check-blanket-noqa
+      - id: python-use-type-annotations
```

Comment:
Presumably, if we *didn't* include this section, then the other code formatting changes that we're seeing on this PR wouldn't be required?

If we *did* want to introduce pre-commit, then I'd strongly suggest that we start of with a pull request that introduces it, but *doesn't* make any other linting changes.